File: components/interactive-card/flutter/ciata_card.dart
import 'package:flutter/material.dart';

/// Implementação experimental do CMP-0010 Card interativo para Flutter.
class CiataCard extends StatelessWidget {
  CiataCard({
    super.key,
    required String title,
    String? description,
    this.onTap,
  })  : title = title.trim(),
        description = description?.trim() {
    if (this.title.isEmpty) {
      throw ArgumentError.value(title, 'title', 'não pode ser vazio');
    }
  }

  final String title;
  final String? description;
  final VoidCallback? onTap;

  @override
  Widget build(BuildContext context) {
    final normalizedDescription =
        description?.isNotEmpty == true ? description : null;

    final content = Padding(
      padding: const EdgeInsets.all(16),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Text(title, style: Theme.of(context).textTheme.titleMedium),
          if (normalizedDescription != null) ...[
            const SizedBox(height: 6),
            Text(normalizedDescription),
          ],
        ],
      ),
    );

    return Card(
      child: onTap == null
          ? content
          : Semantics(
              button: true,
              child: InkWell(onTap: onTap, child: content),
            ),
    );
  }
}

File: components/interactive-card/android/CiataCard.kt
package org.ciata.ds.card

import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Column
import androidx.compose.material3.Card
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier

/** Implementação experimental do CMP-0010 Card interativo para Jetpack Compose. */
@Composable
fun CiataCard(
    title: String,
    modifier: Modifier = Modifier,
    description: String? = null,
    onClick: (() -> Unit)? = null,
) {
    val normalizedTitle = title.trim()
    val normalizedDescription = description?.trim()?.takeIf { it.isNotEmpty() }

    require(normalizedTitle.isNotEmpty()) { "title não pode ser vazio." }

    val cardModifier = if (onClick != null) {
        modifier.clickable(onClick = onClick)
    } else {
        modifier
    }

    Card(modifier = cardModifier) {
        Column {
            Text(normalizedTitle)
            normalizedDescription?.let { Text(it) }
        }
    }
}
